secret-provider/gcp-security-manager/src/main/java/io/camunda/connector/runtime/cloud/CloudSecretProvider.java

```
@@ -0,0 +1,77 @@
+package io.camunda.connector.runtime.cloud;
+
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.camunda.connector.api.secret.SecretProvider;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CloudSecretProvider implements SecretProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CloudSecretProvider.class);
+  private static final Type MAP_TYPE = new TypeToken<Map<String, String>>() {}.getType();
+  public static final String SECRETS_ENV_NAME = "CONNECTOR_SECRETS";
+  public static final String SECRETS_PROPERTY_NAME = "connector.secrets";
+  public static final String SECRETS_PROJECT_ENV_NAME = "SECRETS_PROJECT_ID";
+  public static final String SECRETS_PREFIX_ENV_NAME = "SECRETS_PREFIX";
+
+  private final Map<String, String> secrets;
+
+  public CloudSecretProvider(final Gson gson, final String clusterId) {
+    final String json =
+        Optional.ofNullable(clusterId)
+            .map(CloudSecretProvider::loadGoogleSecrets)
+            .or(CloudSecretProvider::loadEnvironmentSecrets)
+            .or(CloudSecretProvider::loadPropertiesSecrets)
+            .orElse("{}");
```

I think we need to fix this. Either the secrets are loaded from the Secret Manager or this fails.

---

This is a good hint. Actually it does not make sense overall for the CloudSecretProvider to load secrets from anywhere but Google Cloud Secret Manager. I will remove the other  options and let it fail it cannot get things from Google. 

When I want to test something without Google but Environment Variables, I should use a different SecretProvider.